Apps/WebClient/src/ClientApp/src/components/report/immunizationHistory.vue

```diff
@@ -0,0 +1,342 @@
+<script lang="ts">
+import Vue from "vue";
+import { Component, Prop, Ref } from "vue-property-decorator";
+import moment from "moment";
+import container from "@/plugins/inversify.config";
+import { SERVICE_IDENTIFIER } from "@/plugins/inversify";
+import { ResultType } from "@/constants/resulttype";
+import { Action, Getter } from "vuex-class";
+import ImmunizationModel from "@/models/immunizationModel";
+import { DateWrapper, StringISODate } from "@/models/dateWrapper";
+import RequestResult from "@/models/requestResult";
+import PatientData from "@/models/patientData";
+import User from "@/models/user";
+import { ILogger, IImmunizationService } from "@/services/interfaces";
+import BannerError from "@/models/bannerError";
+import ErrorTranslator from "@/utility/errorTranslator";
+import LoadingComponent from "@/components/loading.vue";
+import html2pdf from "html2pdf.js";
+import PDFDefinition from "@/plugins/pdfDefinition";
+
+const userNamespace = "user";
+
+@Component({
+    components: {
+        LoadingComponent,
+    },
+})
+export default class ImmunizationHistoryReportComponent extends Vue {
+    @Prop() private name!: string;
+    @Getter("user", { namespace: userNamespace })
+    private user!: User;
+    @Action("addError", { namespace: "errorBanner" })
+    private addError!: (error: BannerError) => void;
+    @Action("getPatientData", { namespace: userNamespace })
+    getPatientData!: (params: {
+        hdid: string;
+    }) => Promise<RequestResult<PatientData>>;
+    @Ref("report")
+    readonly report!: HTMLElement;
+
+    private logger!: ILogger;
+    private isLoading = false;
+    private notFoundText = "Not Found";
+    private records: ImmunizationModel[] = [];
+    private isPreview = true;
+    private phn = "";
+    private dateOfBirth = "";
+
+    private fetchPatientData() {
+        var patientDataPromise = this.getPatientData({
+            hdid: this.user.hdid,
+        });
+        Promise.all([patientDataPromise])
```

Comment:
only one promise, no need for using Promise.all

Reply:
good point